app/models/user.rb

```diff
@@ -4,21 +4,24 @@ class User < ActiveRecord::Base
   devise :database_authenticatable, :registerable, :confirmable,
          :recoverable, :rememberable, :trackable, :validatable
 
-  has_many :pending_purchases
+  has_many :line_items
+  has_one :cart
+
+  after_save :create_cart
+
+  def create_cart
+    Cart.create!(user: self)
```

Comment:
Cool. Similar to the array thing, I look at that and think, "That might cause your whole app to throw a 500 if something went wrong!" It's a little safer to use 'create' and then handle the failure case however you want.